synfig-core/src/tool/joblistprocessor.cpp

```diff
@@ -93,224 +90,232 @@ std::string replace_extension(const std::string &filename, const std::string &ne
-			job.target->set_alpha_mode(job.alpha_mode);
-		}
+	// If the target type is not yet defined,
+	// try to figure it out from the outfile.
+	if (job.target_name.empty() && !job.outfilename.empty()) {
+		try_to_determine_target_from_outfile(job);
 	}
 
-	// Set the threads and render engine for the target
-	if (job.target)
-	{
-		if(auto scanline_target = Target_Scanline::Handle::cast_dynamic(job.target))
-		{
-			scanline_target->set_threads(SynfigToolGeneralOptions::instance()->get_threads());
-			scanline_target->set_engine(job.render_engine);
-		} else if(auto tile_target = Target_Tile::Handle::cast_dynamic(job.target))
-		{
-			tile_target->set_threads(SynfigToolGeneralOptions::instance()->get_threads());
-			tile_target->set_engine(job.render_engine);
-		}
+	// If the target type is STILL not yet defined, then
+	// set it to a some sort of default
+	set_default_target(job);
+
+	// If no output filename was provided, then create a output filename
+	// based on the given input filename and the selected target.
+	// (ie: change the extension)
+	create_output_filename(job);
+
+	if (!check_permissions(job)) {
+		return false;
+	}
+
+	if (!create_target(job, target_parameters)) {
+		return false;
 	}
 
+	// Set the Canvas on the Target
+	set_canvas_quality_and_alpha_mode(job);
+
+	// Set the threads and render engine for the target
+	set_target_engine_and_threads(job);
+
 	return true;
 }
 
-void process_job (Job& job)
-{
+void print_job_info(const Job& job) {
 	VERBOSE_OUT(3) << job.filename.c_str() << " -- " << std::endl;
-	synfig::info("\tw: %d, h: %d, a: %d, pxaspect: %f, imaspect: %f, span: %f", 
-		job.desc.get_w(), job.desc.get_h(), job.desc.get_antialias(),
-		job.desc.get_pixel_aspect(), job.desc.get_image_aspect(), job.desc.get_span());
-	/*VERBOSE_OUT(3) << '\t'
-				   << boost::format("w:%d, h:%d, a:%d, pxaspect:%f, imaspect:%f, span:%f")
-                                    % job.desc.get_w()
-                                    % job.desc.get_h()
-                                    % job.desc.get_antialias()
-                                    % job.desc.get_pixel_aspect()
-                                    % job.desc.get_image_aspect()
-                                    % job.desc.get_span()
-                    << std::endl;*/
+	synfig::info("\tw: %d, h: %d, a: %d, pxaspect: %f, imaspect: %f, span: %f",
+				 job.desc.get_w(), job.desc.get_h(), job.desc.get_antialias(),
+				 job.desc.get_pixel_aspect(), job.desc.get_image_aspect(), job.desc.get_span());
 
 	synfig::info("\ttl: [%f,%f], br: [%f,%f], focus: [%f,%f]",
-				job.desc.get_tl()[0], job.desc.get_tl()[1],
-				job.desc.get_br()[0], job.desc.get_br()[1],
-				job.desc.get_focus()[0], job.desc.get_focus()[1]
+				 job.desc.get_tl()[0], job.desc.get_tl()[1],
+				 job.desc.get_br()[0], job.desc.get_br()[1],
+				 job.desc.get_focus()[0], job.desc.get_focus()[1]
 	);
+}
+
+void save_canvas_to_file(const std::string& filename, const synfig::Canvas::Handle& canvas) {
```

Comment:
This as well? I am not sure if making these static results in any runtime benefits, but I guess just to stay consistent this can be marked as static.

Reply:
Done